src/EditorFeatures/CSharpTest/Ambiguity/AmbiguousTypeTests.cs

```diff
@@ -0,0 +1,525 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Ambiguity;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Ambiguity
+{
+    public class AmbiguousTypeTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderAndFixer(Workspace workspace)
+            => (null, new CSharpAmbiguousTypeCodeFixProvider());
+
+        private string GetAmbiguousDefinition(string typeDefinion)
+            => $@"
+namespace N1
+{{
+    { typeDefinion }
+}}
+namespace N2
+{{
+    { typeDefinion }
+}}";
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAliasAmbiguousType)]
+        public async Task TestAmbiguousClassObjectCreationUsingsInNamespace()
+        {
+            var classDef = GetAmbiguousDefinition("public class Ambiguous { }");
+            var initialMarkup = classDef + @"
+namespace Test
+{
+    using N1;
+    using N2;
+    class C
+    {
+        void M()
+        {
+            var a = new [|Ambiguous|]();
+        }
+    }
+}";
+            var expectedMarkupTemplate = classDef + @"
+namespace Test
+{
+    using N1;
+    using N2;
+    #
+
+    class C
+    {
+        void M()
+        {
+            var a = new Ambiguous();
+        }
+    }
+}";
+            await TestInRegularAndScriptAsync(initialMarkup, expectedMarkupTemplate.Replace("#", "using Ambiguous = N1.Ambiguous;"), 0);
+            await TestInRegularAndScriptAsync(initialMarkup, expectedMarkupTemplate.Replace("#", "using Ambiguous = N2.Ambiguous;"), 1);
```

Comment:
FWIW, i'm not a fan of this sort of approach.  With code actions, we explicitly spell out the before/after code without the user having to read the test code to figure out how the strings are manipulated.  

This has the benefit of making it very easy to see and grok what will happen without any intermediary steps to understand.

Reply:
nit: named param for 0 and 1.